AirSkyBoat/AirSkyBoat
# Dark Rider Implementation

**_I affirm:_**


## Please enter a player-facing description
No player facing changes

## What does this pull request do? (Please be technical)
Implements Dark Rider behavior based from limited testimonies and information online.

- Dark Rider will spawn every 32-36 hours at night in either Wajaom, Bhaflau, Caedarva, or Mount Z
- After Dark Rider has roamed through the entire zone, he will despawn and simultaneously spawn a warhorse hoofprint. This hoofprint will be interactable if player's are on the appropriate quest to advance their Mercenary Rank. Moreover, this hoofprint will despawn on server midnight. Information about the behavior of when this hoofprint despawns varies greatly online. This implementation is also flexible and can be changed when further evidence is discovered.
- Dark Rider will use Zantetsuken rapidly, as well as instant cast death when engaged. This was based loosely from testimonies found online.
- Zantetsuken was implemented as expected to be seen in the odin BCNM itself.
- Dark Rider will spawn with minions that are spread out along his chosen path. This was based from various testimonies and videos found online.
- Removed Dark Rider's DRK spell list
- Add immunity to everything as well as strong damage resistances to make Dark Rider near impossible

- Dark Rider logic handling is done in Aht_Urghan_Whitegate rather than in each zone to ensure multiple dark riders aren't up at once. It makes use of server variables to handle the logic and to persist through server maintenance
- Created a variable in main settings to control whether or not Dark Rider is used to handle the logic of the hoofprint spawning. 
- Removed widescan from the hoofprints as it appears to have been added in at a later date.

## Some links used
https://ffxiclopedia.fandom.com/wiki/Zantetsuken_Kai
https://www.youtube.com/watch?v=41_1E5ukiUA
https://www.youtube.com/watch?v=vRDFAIS5924
![image](https://github.com/AirSkyBoat/AirSkyBoat/assets/62350957/7087aa61-b3bd-42d5-848f-27fef35c701d)
![image](https://github.com/AirSkyBoat/AirSkyBoat/assets/62350957/a2360fc6-5e7b-47d0-9f2c-e13c53e3ceea)
https://ffxiclopedia.fandom.com/wiki/Warhorse_Hoofprint

## Steps to test these changes
- There is a print debug on line 49. This is useful when testing on local builds to determine where Dark Rider will spawn. Moreover in a singleplayer environment, Dark Rider won't start his pathing until a player enters and initializes the zone.
- For quicker testing, remove the setting of dark rider's respawn time in onRiderDespawn. You can also remove the check to ensure it's night time in order to spawn Dark Rider
